lib/semian/simple_enum.rb

```diff
@@ -0,0 +1,29 @@
+module Semian
+  module Simple
+    class Enum #:nodoc:
+      def initialize(symbol_list:)
+        @allowed_symbols = symbol_list
+        reset
+      end
+
+      attr_reader :value
+
+      def value=(sym)
+        raise ArgumentError unless @allowed_symbols.include?(sym)
+        @value = sym
+      end
+
+      def increment(val = 1)
+        @value = @allowed_symbols[(@allowed_symbols.index(@value) + val) % @allowed_symbols.size]
+      end
```

Comment:
Is this actually used? The only reason I suggested it before was because you inherited from the integer, since you don't do that anymore, I don't think you actually use this.